source/extensions/filters/http/transformation/transformation_filter.cc

```diff
@@ -205,6 +209,32 @@ void TransformationFilter::addEncoderData(Buffer::Instance &data) {
   encoder_callbacks_->addEncodedData(data, false);
 }
 
+void TransformationFilter::transformOnStreamCompletion() {
+  if (on_stream_completion_transformation_ == nullptr) {
+    return;
+  }
+
+  Http::StreamFilterCallbacks *on_complete_callbacks{};
+  // Body isn't required for this transformer since it isn't included
+  // in access logs
+  Buffer::OwnedImpl emptyBody{};
+  try {
+    on_stream_completion_transformation_->transform(*response_headers_,
+                                                    request_headers_, 
+                                                    emptyBody, 
+                                                    *on_complete_callbacks);
+    on_stream_completion_transformation_->transform(*request_headers_,
+                                                    request_headers_, 
+                                                    emptyBody, 
+                                                    *on_complete_callbacks);
```

Comment:
I think I see why you're calling twice, but I think using the existing `transform` function is proving to be more complicated than it's worth. Can we instead create a new function to just pass in the necessary data 1 time. 
```
    on_stream_completion_transformation_->transformOnStreamComplete(*request_headers_,
                                                    *response_headers_, 
                                                    *on_complete_callbacks);
```
Or something like that

wdyt @yuval-k

Reply:
Yea, I was thinking the same. It should simplify this if I just had a function that took in everything I needed.